src/sortBy.ts

```diff
@@ -1,29 +1,59 @@
 import { purry } from './purry';
 
+type Direction = 'asc' | 'desc';
+type SortProjection<T> = (x: T) => Sortable;
+type SortablePrimitive = number | string;
+type Sortable = SortablePrimitive | { valueOf(): SortablePrimitive };
+type SortPair<T> = [SortProjection<T>, Direction];
+type SortRule<T> = SortProjection<T> | SortPair<T>;
+
 /**
- * Sorts the list according to the supplied function in ascending order.
+ * Sorts the list according to the supplied functions and directions.
  * Sorting is based on a native `sort` function. It's not guaranteed to be stable.
+ *
+ * Directions are applied to functions in order and default to ascending if not specified.
  * @param array the array to sort
- * @param fn the mapping function
+ * @param sorts a list of mapping functions and optional directions
  * @signature
- *    R.sortBy(array, fn)
+ *    R.sortBy(array, ...sorts)
  * @example
  *    R.sortBy(
  *      [{ a: 1 }, { a: 3 }, { a: 7 }, { a: 2 }],
  *      x => x.a
  *    )
  *    // => [{ a: 1 }, { a: 2 }, { a: 3 }, { a: 7 }]
+ *
+ *    R.sortBy(
+ *     [
+ *       {color: 'red', weight: 2},
+ *       {color: 'blue', weight: 3},
+ *       {color: 'green', weight: 1},
+ *       {color: 'purple', weight: 1},
+ *     ],
+ *      [x => x.weight, 'asc'], x => x.color
```

Comment:
I wasn't aware that we would _also_ allow for mixing the two paradigms - tuples and / or functions, but if it works it's nice. I don't see that we have a test for this mixture though, can you maybe add one?